include/boost/geometry/core/cs.hpp

```diff
@@ -107,12 +107,8 @@ known as lat,long or lo,la or phi,lambda
 */
 template<typename DegreeOrRadian>
 struct geographic
-{
-    typedef typename core_detail::coordinate_system_units
-        <
-            DegreeOrRadian
-        >::units units;
-};
+    : core_detail::define_angular_units<DegreeOrRadian>
+{};
```

Comment:
What is the reason to derive instead of defining it inside?

Reply:
Less lines of code. AFAIU it's a typical way of defining MPL-like meta-functions.